src/python/gudhi/representations/vector_methods.py

```diff
@@ -719,21 +721,26 @@ class Atol(BaseEstimator, TransformerMixin):
     >>> a = np.array([[1, 2, 4], [1, 4, 0], [1, 0, 4]])
     >>> b = np.array([[4, 2, 0], [4, 4, 0], [4, 0, 2]])
     >>> c = np.array([[3, 2, -1], [1, 2, -1]])
-    >>> atol_vectoriser = Atol(quantiser=KMeans(n_clusters=2, random_state=202006))
+    >>> atol_vectoriser = Atol(quantiser=KMeans(n_clusters=2, random_state=202006, n_init=10))
     >>> atol_vectoriser.fit(X=[a, b, c]).centers
     array([[ 2.6       ,  2.8       , -0.4       ],
            [ 2.        ,  0.66666667,  3.33333333]])
-    >>> atol_vectoriser(a)
+    >>> atol_vectoriser._transform(a)
     array([0.42375966, 1.18168665])
-    >>> atol_vectoriser(c)
+    >>> atol_vectoriser._transform(c)
     array([1.25157463, 0.02062512])
     >>> atol_vectoriser.transform(X=[a, b, c])
     array([[0.42375966, 1.18168665],
            [1.06330156, 0.29861028],
            [1.25157463, 0.02062512]])
     """
     # Note the example above must be up to date with the one in tests called test_atol_doc
-    def __init__(self, quantiser, weighting_method="cloud", contrast="gaussian"):
+    def __init__(
+            self,
+            quantiser=KMeans(n_clusters=2, random_state=202312, n_init=10),
```

Comment:
For the tests (and the values are repeated there), I understand that we want a fixed random_state so the output is deterministic, only 2 clusters because there are very few points, and `n_init=10` because that's what we had before. But I wonder if those are the best choices as default values (for comparison, plain `KMeans()` with all default arguments would use 8, None and 'auto' instead). Do we have a reason to believe that greedy k-means++ won't work well on diagrams and that we need more iterations (`n_init`) than sklearn would do by default? Do we have a reason to recommend a very low number of clusters (2)? Do we have a reason to force determinism?

Reply:
I put a `n_init=10` to suppress the warning sklearn had for not setting the parameter, and because `n_init=10` is very understandable contrary to `n_init='auto'`. I recommend setting either, because it can be bothersome on some upper levels if a lower code produces warning on a regular basis... Or we could not set the parameter and hide the warning...
I put `n_clusters=2` because its the lightest we can use that is still multidimensional, so because it can suffice and one should probably work upwards from 2. We could leave the default here.
And no reason to force determinism outside of tests, we can definitely remove `random_state`!

Reply:
> I put a `n_init=10` to suppress the warning sklearn had for not setting the parameter, and because `n_init=10` is very understandable contrary to `n_init='auto'`. I recommend setting either, because it can be bothersome on some upper levels if a lower code produces warning on a regular basis... Or we could not set the parameter and hide the warning...

Since sklearn considers `auto` to be the right value (and in particular it yields `1` for the default k-means++), I think it would be nicer to use that (explicitly to avoid a warning with sklearn 1.[23]). This means requiring sklearn 1.2+, I don't know if we already do that.

> I put `n_clusters=2` because its the lightest we can use that is still multidimensional, so because it can suffice and one should probably work upwards from 2.

ok. (I may add a word to the doc to encourage people to override the default)

> And no reason to force determinism outside of tests, we can definitely remove `random_state`!

:+1: